schema/policy.go

```diff
@@ -413,6 +430,59 @@ func (p *Policy) AddTenantToFilter(action string, tenant Tenant) {
 	p.actionTenantFilter[action] = append(p.actionTenantFilter[action], tenant)
 }
 
+// AddPropertyConditionFilter adds property based filter for action
+func (p *Policy) AddPropertyConditionFilter(action string, match map[string]interface{}) {
+	p.actionPropertyConditionFilter[action] = append(p.actionPropertyConditionFilter[action], match)
+}
+
+// ApplyPropertyConditionFilter applies filter based on Property
+func (p *Policy) ApplyPropertyConditionFilter(action string, data map[string]interface{}, next map[string]interface{}) error {
```

Comment:
It is unclear to me what the `next` argument represents and how it is used. It seems that most of the times it is not used... only on update ? could you have just called this same method for the map of properties that you are going to update ? why is next applied to map but not to list or item cases.